content/getting-started/swift.textile

```diff
@@ -0,0 +1,285 @@
+
+h2(#prerequisites). Prerequisites
+
+* Sign up for an Ably account.
+** Create a new app, and create your first API key.
+** Your API key will need the @publish@, @subscribe@, @presence@ and @history@ capabilities.
+
+* Install the Ably CLI:
+
+```[sh]
+npm install -g @ably/cli
+```
+
+* Run the following to log in to your Ably account and set the default app and API key:
+
+```[sh]
+ably login
+
+ably apps switch
+ably auth keys switch
+```
+
+* Install "Xcode":https://developer.apple.com/xcode/
+
+* Install the Ably SDK in your Swift project using Swift Package Manager:
+
+```[swift]
+dependencies: [
+    .package(url: "https://github.com/ably/ably-cocoa", from: "1.2.20")
+]
+```
+
+
+<aside data-type='note'>
+<p>The code examples in this guide include a demo API key. If you wish to interact with the Ably CLI and view outputs within your Ably account, ensure that you replace them with your own API key.</p>
+<p><strong>Important:</strong> API keys are suitable for development and demo purposes only. For production apps, <a href="/docs/authentication/token-authentication">use token authentication</a> to keep your credentials secure.</p>
+</aside>
+
+h2(#xcode-setup). Running this example in Xcode (SwiftUI)
+
+To follow this guide in Xcode, use a new iOS project with the SwiftUI App template. All code can be added directly to your @ContentView.swift@ file, inside the @ContentView@ struct. Use the @.onAppear@ modifier to run the Ably code when the view appears. No additional files or setup are needed.
+
+*Steps:*
+
+# Open Xcode and create a new iOS project (App template, Swift, SwiftUI).
+# Open @ContentView.swift@.
+# Add the code snippets from each step inside the @.onAppear@ block in the @ContentView@ struct.
+# Run the app using the Simulator or a real device.
+# All @print@ output will appear in Xcode's debug console (View > Debug Area > Activate Console, or press Cmd+Shift+C).
+
+h2(#step-1). Step 1: Connect to Ably
+
+Clients establish a connection with Ably when they instantiate an SDK. This enables them to send and receive messages in realtime across channels.
+
+* Open up the "dev console":https://ably.com/accounts/any/apps/any/console of your first app before instantiating your client so that you can see what happens.
+
+* Add these imports to the top of your Swift file:
+
+```[swift]
+import Ably
+```
+
+* Create an instance of @ARTRealtime@ with your API key. This is the main entry point for the realtime library. The @ARTRealtime@ client connects to Ably as soon as it's instantiated, using a basic transport such as a raw TCP socket or a WebSocket:
+
+```[swift]
+let realtime = ARTRealtime(key: "{{API_KEY}}")
+```
+
+* You can monitor the connection state by adding a listener to the connection property:
+
+```[swift]
+realtime.connection.on { stateChange in
+    print("Connection state changed to: \(stateChange.current)")
+    
+    if stateChange.current == .connected {
+        print("Made my first connection!")
```

Comment:
On one hand, I think we shouldn't overcomplicate things; on the other hand, printing to the console in a SwiftUI project feels a bit out of place. Maybe we can create simple in-app console (just render list of lines in a text box)? Or maybe create console desktop swift app instead of swift ui?